.github/workflows/docker.yml

```diff
@@ -0,0 +1,50 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'push' && github.run_number || github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  push-docker-image:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Get latest version
+        id: get-latest-version
+        run: echo "::set-output name=version::$(ls -t build-specs | head -n1 | grep -o -E '[0-9]+\.[0-9]+\.[0-9]+')"
```

Comment:
::set-output is an old command and it will stop working soon

you have to send the output like

`echo "stuff" >> $GITHUB_OUTPUT`